huggingface_pipelines/dataset.py

```diff
@@ -0,0 +1,67 @@
+from typing import TypedDict
+from dataclasses import dataclass, field, replace
+import uuid
+
+
+class DatasetOverwrites(TypedDict, total=False):
+    dataset_name: str
+    dataset_split: str
+    world_size: int
+    rank: int
+    cache_dir: str
+
+
+@dataclass
+class DatasetConfig():
+    """
+    Configuration class for loading and sharding datasets.
+
+    Attributes:
+        dataset_name (str): The name of the dataset to load.
+        dataset_split (str): The split of the dataset to use (e.g., 'train', 'test', 'validation').
+        world_size (int): The number of shards to split the dataset into. Defaults to 1.
+        rank (int): The ID of the shard to retrieve. Defaults to 0.
+        cache_dir (str): The directory to cache the loaded dataset. Defaults to None.
+        trust_remote_code (bool): Whether to trust remote code when loading the dataset. Defaults to False.
+    """
+    dataset_name: str
+    dataset_split: str
+    config: str = None
+    world_size: int = 1
+    rank: int = 0
+    cache_dir: str = None
+    trust_remote_code: bool = False
+    uuid: str = field(default_factory=lambda: str(uuid.uuid4()))
+    output_dir: str = "results"
```

Comment:
this could be None in some case and let's not put "results"  as default (usually this will an absolute path and in many case we'll not write the intermediate results at all)